controllers/email.js

```diff
@@ -0,0 +1,42 @@
+const nodemailer = require("nodemailer");
+const sgMail = require("@sendgrid/mail");
+const ejs = require("ejs");
+const path = require("path");
+require("dotenv").config();
+
+sgMail.setApiKey(process.env.SENDGRID_API_KEY);
+
+const transporter = nodemailer.createTransport({
+  service: "SendGrid",
+  auth: {
+    user: "apikey",
+    pass: process.env.SENDGRID_API_KEY,
+  },
+});
+
+module.exports = {
+  sendVerificationEmail: async (email, verificationToken, req) => {
+    try {
+      const verificationLink = `${req.protocol}://${req.get(
+        "host"
+      )}/api/users/verify/${verificationToken}`;
+
+      const html = await ejs.renderFile(
+        path.join(__dirname, "../../views/verificationEmail.ejs"),
+        { verificationLink }
+      );
+
+      const mailOptions = {
+        to: email,
+        from: process.env.SENDER_EMAIL,
```

Comment:
##### Generyczna funkcja wysyłania email
Dobra robota z dotychczasową implementacją! Jednakże, warto by było uczynić funkcję odpowiedzialną za wysyłanie wiadomości bardziej generyczną. To pozwoli na jej łatwiejsze wykorzystanie w różnych częściach aplikacji, a także ułatwi modyfikację i rozbudowę w przyszłości.
Spróbuj dodać parametry do funkcji, takie jak adresat, temat, treść wiadomości i inne opcje konfiguracyjne, które mogą być potrzebne w zależności od kontekstu użycia. Dzięki temu będziesz mogła używać tej samej funkcji do wysyłania różnych rodzajów wiadomości bez konieczności tworzenia nowej funkcji za każdym razem, gdy potrzebujesz wysłać coś innego.